lib/charms/layer/__init__.py

```diff
@@ -42,3 +22,21 @@ def import_layer_libs():
         ):
             continue
         import_module('charms.layer.{}'.format(module_name))
+
+
+# Terrible hack to support the old terrible interface.
+# Try to get people to call layer.options.get() instead so
+# that we can remove this garbage.
+# Cribbed from https://stackoverfLow.com/a/48100440/4941864
+class OptionsBackwardsCompatibilityHack(sys.modules[__name__].__class__):
+    def __call__(self, section=None, layer_file=None):
+        if layer_file is None:
+            return self.get(section=section)
+        else:
+            return self.get(section=section,
+                            layer_file=Path(layer_file))
+
+
+def patch_options_interface():
+    from charms.layer import options
+    options.__class__ = OptionsBackwardsCompatibilityHack
```

Comment:
Can this not be done immediately? I'm worried there might be some action out there that imports and uses `layer.options` but never calls `activate_venv`

Reply:
No, because the yaml dep in the options layer means we can't import it before at a minimum of pyyaml is installed, which happens early during bootstrap.  The patching is done during the bootstrap / startup logic (in `activate_venv`) before any other charm code is run (except the `execd` stuff, which knows it has to not have any dependencies).